rest/config.go

```diff
@@ -1307,6 +1307,17 @@ func (sc *ServerContext) _applyConfig(cnf DatabaseConfig, failFast bool) (applie
 		}
 	}
 
+	if sc.bootstrapContext.connection != nil {
```

Comment:
When is bootstrapContext.connection == nil?  Are we skipping this check when running with file-based config?

Reply:
I decided to skip this in file based config because unlike in 3.1, we don't have a bootstrap connection in file based config, so I'd have to make and close a bucket specific connection. Since the primary consumer of this feature is capella, I thought that I'd avoid adding code complexity for a minor gain.

It is checked in 3.1 for file based config.